snappi_ixnetwork/device/macsec.py

```diff
@@ -0,0 +1,333 @@
+        secy = ethernet_interface.get("secy")
+        self._config_secy_engine_encryption_only_tx_pn(secy, ixn_staticmacsec)
+        self._config_secy_engine_encryption_only_traffic(device, ethernet_interface, ixn_staticmacsec)
+
+    def _config_secy_engine_encryption_only_traffic(self, device, ethernet_interface, ixn_staticmacsec):
+        ethernet_name = ethernet_interface.get("eth_name")
+        secy = ethernet_interface.get("secy")
+        self.logger.debug("Configuring stateless encryption traffic from ethernet %s secy %s" % (ethernet_name, secy.get("name")))
+        ethernets = device.get("ethernets")
+        for ethernet in ethernets:
+            if ethernet.get("name") == ethernet_name:
+                ipv4_addresses = ethernet.get("ipv4_addresses")
+                if ipv4_addresses is None:
+                    raise Exception("IPv4 not configured on ethernet %s" % ethernet_name)
+                elif len(ipv4_addresses) > 1:
+                    raise Exception("More than one IPv4 address configured on ethernet %s" % ethernet_name)
+                ipv4_address = ipv4_addresses[0].address
+                ipv4_gateway_mac = ipv4_addresses[0].gateway_mac
+                if ipv4_gateway_mac is None:
+                    raise Exception("IPv4 gateway mac not configured for IPv4 address %s on ethernet %s" % (ethernet_name, ipv4_address))
+                elif not ipv4_gateway_mac.choice == "value" or ipv4_gateway_mac.value is None:
+                    raise Exception("IPv4 gateway static mac not configured for IPv4 address %s on ethernet %s" % (ethernet_name, ipv4_address))
+                ipv4_gateway_static_mac = ipv4_addresses[0].gateway_mac.value
+                ixn_staticmacsec["sourceIp"] = self.multivalue(ipv4_address)
+                ixn_staticmacsec["dutMac"] = self.multivalue(ipv4_gateway_static_mac)
+                break
+
+    def _config_secy_engine_encryption_only_tx_pn(self, secy, ixn_staticmacsec):
+        engine_tx_pn = secy.crypto_engine.stateless_encryption_only.tx_pn
+        if engine_tx_pn is None:
+            return
+        self.logger.debug("Configuring Tx PN of stateless encryption only engine secy %s" % secy.get("name"))
+        if engine_tx_pn.choice == "fixed_pn":
+            ixn_staticmacsec["incrementingPn"] = False
+            engine_tx_pn_fixed = engine_tx_pn.fixed
+            ixn_staticmacsec["fixedPn"] = engine_tx_pn_fixed.pn
+            ixn_staticmacsec["mvFixedXpn"] = self.multivalue(engine_tx_pn_fixed.xpn)
+        elif engine_tx_pn.choice == "incrementing_pn":
+            ixn_staticmacsec["incrementingPn"] = True
+            engine_tx_pn_incr = engine_tx_pn.incrementing
+            ixn_staticmacsec["packetCountPn"] = engine_tx_pn_incr.count
+            ixn_staticmacsec["firstPn"] = engine_tx_pn_incr.first_pn
+            ixn_staticmacsec["mvFirstXpn"] = self.multivalue(engine_tx_pn_incr.first_xpn)
```

Comment:
not found any code to change card mode to masec otherwise user will raise issue if we want their card should be converted to macsec mode.